python/ray/cluster_utils.py

```diff
@@ -128,6 +128,17 @@ def remove_node(self, node, allow_graceful=True):
             node (Node): Worker node of which all associated processes
                 will be removed.
         """
+        global_node = ray.worker._global_node
+        if global_node is not None:
+            if node._raylet_socket_name == global_node._raylet_socket_name:
```

Comment:
Comparing socket name should be sufficient because cluster_util will only run in a single node.